enterprise/cypher/slotted-runtime/src/main/java/org/neo4j/cypher/internal/compatibility/v3_4/runtime/slotted/DefaultComparatorTopTable.java

```diff
@@ -126,7 +126,8 @@ public T next()
                 {
                     throw new NoSuchElementException();
                 }
-                return (T) array[--cursor];
+                --cursor;
+                return (T) array[cursor];
```

Comment:
I don't see how this is an improvement.

Reply:
Same reason as the increment operator in CompiledConversionUtils

Reply:
I think the new version is slightly more confusing. A less confusing middle-ground would be:
```
int offset = --cursor;
return (T) array[offset];
```